server_xl_test.go

```diff
@@ -1482,3 +1587,114 @@ func (s *MyAPIXLSuite) TestObjectMultipartOverwriteSinglePut(c *C) {
 	c.Assert(n, Equals, int64(len([]byte("hello world"))))
 	c.Assert(true, Equals, bytes.Equal(buffer3.Bytes(), []byte("hello world")))
 }
+
+func (s *MyAPIXLSuite) TestMultipleObjectsOverlappingPath(c *C) {
+	// Put object /a/b/c/d, should succeed
+	buffer1 := bytes.NewReader([]byte("hello one"))
+	request, err := s.newRequest("PUT", testAPIXLServer.URL+"/multipleobjects/a/b/c/d", int64(buffer1.Len()), buffer1)
+	c.Assert(err, IsNil)
+
+	client := http.Client{}
+	response, err := client.Do(request)
+	c.Assert(err, IsNil)
+	bod, err := ioutil.ReadAll(response.Body)
+	fmt.Println(string(bod))
+	c.Assert(response.StatusCode, Equals, http.StatusOK)
+
+	request, err = s.newRequest("GET", testAPIXLServer.URL+"/multipleobjects/a/b/c/d", 0, nil)
+	c.Assert(err, IsNil)
+
+	client = http.Client{}
+	response, err = client.Do(request)
+	c.Assert(err, IsNil)
+	c.Assert(response.StatusCode, Equals, http.StatusOK)
+
+	// verify response data
+	responseBody, err := ioutil.ReadAll(response.Body)
+	c.Assert(err, IsNil)
+	c.Assert(true, Equals, bytes.Equal(responseBody, []byte("hello one")))
+
+	// Put object a/b/c, should fail because it is a directory
+	buffer2 := bytes.NewReader([]byte("hello two"))
+	request, err = s.newRequest("PUT", testAPIXLServer.URL+"/multipleobjects/a/b/c", int64(buffer2.Len()), buffer2)
+	c.Assert(err, IsNil)
+
+	client = http.Client{}
+	response, err = client.Do(request)
+	c.Assert(err, IsNil)
+	verifyError(c, response, "InternalError", "We encountered an internal error, please try again.", http.StatusInternalServerError)
```

Comment:
@krishnasrinivas This request should result in an error but it is successful. The same test works on fs, but on xl we are getting unexpected results: success for small/large put object a/b/c when an object a/b/c/d is present.